### ISSUE TITLE
Check SiS configurations for invalid certificates

### ISSUE THREAD

**Issue by joeniquette** [department-of-veterans-affairs/va.gov-team#82171]
We need to automatically check certificates stored in client and service account configurations for expired, expiring, and self-signed certificates.

### Expiring/Expired
automate a task that will notify the identity team when a cert integrated with SiS is going to expire. These are usually certs that other teams are going to be created by other teams but we want to ensure we notify them before they expire.

### Self-signed
Both Sign in Service client auth private key jwt and service account should confirm that the public certs they use as part of authentication are not locally signed. These should be signed by an approved entity. If on dev/staging/prod, and the cert is locally signed, we should periodically send a message to data dog that we can monitor and ensure we have no certs that are locally signed in these stacks

**Comment by dickdavis** [comment]
Finished initial implementation of daily certificate checking job and put in for review.

**Comment by dickdavis** [comment]
Combined multiple tickets.

**Comment by dickdavis** [comment]
I completed a functioning prototype for how we can accomplish this functionality and provided it to the team for review, which is available [here](https://github.com/department-of-veterans-affairs/vets-api/pull/16733).

After discussing it, we determined that the current implementation of how certificates are stored for configurations is suboptimal. The current implementation relies on storing objects as an array of strings containing the plaintext version of the certificate. A more ideal implementation would be to store these certificates as individual records with associations to their respective configuration.

We determined the proposed refactor would yield a more maintainable and performant system with less code. This is in part due to the strengths of using an ORM-backed data model instead of relying on Ruby primitives.

Given this, I've closed this ticket and instead [created an epic](https://app.zenhub.com/workspaces/identity-5f5bab705a94c9001ba33734/issues/gh/department-of-veterans-affairs/va.gov-team/83214) that captures all of the work necessary to accomplish the refactor and fulfill the original requirements of this ticket.
